### PR TITLE
fix: remove trailing spaces where supported

### LINKED ISSUE #136: Remove trailing spaces before new line
Craft-CLI messages are padded with spaces to fill the entire terminal width. This causes problems when copying and pasting output because the spaces are also copied.

Example:
```
$ snapcraft
Launching instance...                                                                                                                                                                 <-- ends here
Executed: repull my-part ('source' property changed)                                                                                                                                  <-- ends here
Executed: overlay my-part                                                                                                                                                             <-- ends here
Executed: build my-part                                                                                                                                                               
Executed: stage my-part                                                                                                                                                               
Executed: prime my-part                                                                                                                                                               
Executed parts lifecycle                                                                                                                                                              
Generated snap metadata                                                                                                                                                               
Created snap package my-snap-name_0.1_amd64.snap
```

### PR DESCRIPTION
**Pull request by lengau**
Fixes #136
CRAFT-3358

- [x] Have you followed the guidelines for contributing?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `tox`?

### CODE REVIEW

**Review by cmatsuoka — APPROVED** [review]
Nice fix!

File: craft_cli/printer.py [hunk 1]
```diff
@@ -87,9 +109,8 @@ def _format_term_line(prefix: str, text: str, spintext: str, *, ephemeral: bool)
             text = text[-remaining_for_last_line:]
             if len(text) > usable:
                 text = text[: usable - 1] + "…"
-    cleaner = " " * (usable - len(text) % width)
 
-    return prefix + text + spintext + cleaner
+    return prefix + fill_line(text + spintext)
```
Comment (tigarmo) [inline craft_cli/printer.py:113]: don't you need to take the prefix into account, for the non-`ANSI_CLEAR_LINE_TO_END` case?

File: craft_cli/printer.py [hunk 2]
```diff
@@ -71,6 +75,24 @@ def _get_terminal_width() -> int:
...
+    """Whether the current environment supports ANSI escape sequences."""
+    if platform.system() != "Windows":
+        return True
+    return "WT_SESSION" in os.environ  # Windows Terminal supports ANSI escape sequences.
+
+
+def fill_line(text: str) -> str:
+    """Turn the input text into a line that will fill the terminal."""
+    if _supports_ansi_escape_sequences():
+        return text + ANSI_CLEAR_LINE_TO_END
```
[the remark this answers is not in the document]
Comment (lengau) [reply craft_cli/printer.py:89]: The cursor sits at the end of the printed line, but at this point we're supposed to not have a line that's too long for the terminal.

**Comment by tigarmo** [comment]
@lengau I played around with this a bit and it seems to be working fine! That is, no breakage that I could see
One thing that I'd like your opinion on though is the new behavior where the cursor keeps jumping around the row as the line that's printed is shorter or longer; I found that distracting but maybe I'm too close to the issue to get a good sense.

Here's an example: this is snapcraft with the old behavior (cursor at the end of the artificially-inflated line):
cursor-old

And here's rockcraft with the new behavior:
cursor-new

File: craft_cli/printer.py (hunk 1 above)
Comment (lengau) [reply craft_cli/printer.py]: Actually I stand corrected. The `prefix` here is the previous line's end. I've renamed it for less ambiguity.

**Comment by lengau** [comment]
@tigarmo based on our team discussions I've created https://github.com/canonical/craft-cli/pull/275 to go along with this and cover the cursor hiding.

**Review by tigarmo — APPROVED** [review]
🤩

**Pull request merged**